src/hotspot/share/ci/ciReplay.cpp

```diff
@@ -437,6 +440,9 @@ class CompileReplay : public StackObj {
       }
 
       Klass* k = parse_klass(CHECK_NULL);
+      if (k == NULL) {
+        return NULL;
+      }
```

Comment:
Do we need a call to report_error() for a descriptive message for these failure paths?

Reply:
parse_method() has calls to report_error() when it returns NULL. I think it's true of parse_klass() except if SymbolTable::new_symbol() returns NULL (but can it return NULL?). Do you think extra messages make sense then?

Reply:
No, I don't think new_symbol can return NULL.  I think you're good to integrate.